### ISSUE TITLE
Preprocessor - Fails on list of numbers (without [])

### ISSUE THREAD

**Issue by PabstMirror** [LordGolias/sqf#14]
```
#define TAG_COLOR 0.15,0.451,0.94
x = [TAG_COLOR];
```
```
[1,28]:error:can't interpret statement (missing ;?): <Nothing(Nothing)> <Nothing(Nothing)> <Number(0.94)>
```

**Comment by LordGolias** [comment]
Yap. Atm `#define` second argument has to be syntactically correct because it is parsed as code. I would argue that 

```
#define TAG_COLOR [0.15,0.451,0.94]
x = TAG_COLOR;
```

is the most human-readable way of writing the particular example you gave, but I agree that in some cases the construct is useful. 

Like in issue #11, the problem here is that formally there should be a preprocessor that would replace defines *before* the parser parses it. Currently, the defines are replaced *after* the parser, by the analyser. This is what causes the issue here.

**Comment by jonpas** [comment]
That construct is mostly useful when it might be a part of a bigger array, eg. some values + some more run-time calculated values.

**Comment by LordGolias** [comment]
Fixed in e0195cd883520e31650b29617381fe10a22d1f90. Thanks for the report. Let me know if you find any other edge case. This commit was run against latest ACE, ACRE and Exile double-check it does not raise any false positives.
